pkg/controller/messagingproject/messagingproject_controller_test.go

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package messagingproject
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	v1 "github.com/enmasseproject/enmasse/pkg/apis/enmasse/v1"
+	"github.com/enmasseproject/enmasse/pkg/client/clientset/versioned/scheme"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func setup(t *testing.T, ns *corev1.Namespace) *ReconcileMessagingProject {
+	s := scheme.Scheme
+	s.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.Namespace{})
+	objs := []runtime.Object{
+		ns,
+	}
+	cl := fake.NewFakeClientWithScheme(s, objs...)
+	r := &ReconcileMessagingProject{client: cl}
+	return r
+}
+
+func assertSelector(t *testing.T, expected bool, selector *v1.NamespaceSelector) {
+	r := setup(t, &corev1.Namespace{
```

Comment:
I found the fact that `assertSelector` was producing/passing some of the test data pretty confusing at first sight.   I'd move the setup call and extract arg 2 of the r.matchesSelector call.

Reply:
Not sure what you meant with that last sentence, but let me know if the refactoring improves thing.